### ISSUE TITLE
The plugin requires a higher Android SDK version

### ISSUE THREAD

**Issue by AmrF93** [firebase/firebase-android-sdk#6168]
Flutter: 3.22.3
Android Studio version: 2023.2
Firebase Component: Auth (Database, Firestore, Storage, Functions, etc)
Firebase Auth: 5.1.4

I have an error after I added Firebase Auth plugin:

> The plugin firebase_auth requires a higher Android SDK version.                                                 
 Fix this issue by adding the following to the file C:\Users\HP\Desktop\app\test_i\android\app\build.gradle:     │
 android {                                                                                                       
   defaultConfig {                                                                                               
     minSdkVersion 23                                                                                           
   }                                                                                                             
 }                                                                                                               
 Following this change, your app will not be available to users running Android SDKs below 23.                   │
 Consider searching for a version of this plugin that supports these lower versions of the Android SDK instead.

But when I went to build.gradle I found it like this:
``
defaultConfig {
        applicationId = "com.example.test_i"
        minSdk = flutter.minSdkVersion
        targetSdk = flutter.targetSdkVersion
        versionCode = flutterVersionCode.toInteger()
        versionName = flutterVersionName
    }
``

I;ve tried to replace flutter.minSdkVersion with 23, like this:
``
defaultConfig {
        applicationId = "com.example.test_i"
        minSdk =  23
        targetSdk = flutter.targetSdkVersion
        versionCode = flutterVersionCode.toInteger()
        versionName = flutterVersionName
    }
``

But it didn't work either, and gave me another error:
> Launching lib\main.dart on sdk gphone x86 in debug mode...
Running Gradle task 'assembleDebug'...                              7.8s
√ Built build\app\outputs\flutter-apk\app-debug.apk
Installing build\app\outputs\flutter-apk\app-debug.apk...        1,870ms
Error: ADB exited with exit code 1
Performing Streamed Install
adb: failed to install C:\Users\HP\Desktop\app\test_ii\build\app\outputs\flutter-apk\app-debug.apk: Failure
[INSTALL_FAILED_INSUFFICIENT_STORAGE]
Error launching application on sdk gphone x86.

Any help, guys?

**Comment by lehcar09** [comment]
Hi @AmrF93, thank you for reaching out. Could you try cleaning up your build by running `flutter clean` and rebuild your app? 

In addition to that, I recommend modifying the `flutter.minSdkVersion` declared in the local.properties file of your app. To streamline the change.

**Comment by AmrF93** [comment]
Thank you @lehcar09 for your time. I've tried that, but it didn't work. Any other tips.

**Comment by lehcar09** [comment]
Hey @AmrF93, pardon me, just to clarify, does the issue `The plugin firebase_auth requires a higher Android SDK version.` persist after you replace the `flutter.minSdkVersion` with `23`?

Looking into the second error you shared, the APK file was build and that means the minSdkVersion is resolved. 

The second error `Failure [INSTALL_FAILED_INSUFFICIENT_STORAGE]` you encountered is not related with the first error. This error indicates that the Android Debug Bridge (ADB) encountered an issue while trying to install your app on the emulator or device. This could be resolved by freeing up a space on the device or emulator. You can check on this similar issue [#4508](https://github.com/flutter/flutter-intellij/issues/4508) filed in the Flutter repository.

**Comment by AmrF93** [comment]
It was the first problem that led to the second; because before installing the firebase_auth plugin, I had no problem.
I did not find the solution in the topic #4508 you sent me. But it made me realize the new problem. So I looked for other solutions to it. I tried many of them and finally found the solution:
- 1st error: Change "`minSdk = flutter.minSdkVersion`" to "`minSdk = 23`". For me, it was 23, and it's the same SDK version given in the error.
- 2nd error: Increase the internal storage of my emulator through AVD, from 6 to 10 GB.
So, thank you @lehcar09, you have been a great help.
